source/state_representation/README.md

```diff
@@ -821,203 +827,618 @@ The inverse of a state will set the wrench to zero.
 ```
 
-### Matrix multiplication
+Addition and subtraction is supported between `JointTorques` and `JointState`, provided that they are compatible.
 
-The `Jacobian` object contains an underlying `Eigen::MatrixXd` which can be retrieved using the `Jacobian::data()`
-method.
-Direct multiplication of the `Jacobian` object with another `Eigen::MatrixXd` has also been implemented and returns the
-`Eigen::MatrixXd` product.
+The return type of each compatible operation is shown below.
 
 ```c++
-state_representation::Jacobian jac("myrobot", 3, "eef", Eigen::MatrixXd::Random(6, 3));
-// alternatively can use the Random static constructor
-state_representation::Jacobian jac = state_representation::Jacobian::Random("myrobot", 3, "eef");
-Eigen::MatrixXd mat = jac.data();
-Eigen::MatrixXd res = jac * Eigen::MatrixXd(3, 4); // equivalent to  jac.data() * Eigen::MatrixXd(3, 4);
+JointTorques r1 = joint_torques + other_joint_torques;
+JointState r2 = joint_torques + joint_state;
+JointState r3 = joint_state + joint_torques;
+
+JointTorques r4 = joint_torques - other_joint_torques;
+JointState r5 = joint_torques - joint_state;
+JointState r6 = joint_state - joint_torques;
+
+joint_torques += other_joint_torques; // equivalent to joint_torques = joint_torques + other_joint_torques
+joint_torques += joint_state; // equivalent to joint_torques = joint_torques + joint_state
+
+joint_torques -= other_joint_torques;
+joint_torques -= joint_state;
+```
+
+## Jacobian
+
+In robotics, the Jacobian is used to map state variables between joint space and Cartesian space. Mathematically, it
+is a matrix containing the set of partial derivatives for a vector function.
+
+Most commonly, it is used to convert joint velocities of a robot into the Cartesian velocity of the end-effector.
+Conversely, the transpose can map a Cartesian wrench applied at the end-effector to the associated torques at each
+joint. The Jacobian matrix has additional uses for inverse kinematics and null-space control.
+
+The `Jacobian` class is a wrapper for the underlying matrix that works directly with `JointState` and `CartesianState`
+types while providing additional methods and operators.
+
+### Reference frame and joint names
+
+Because the `Jacobian` class is designed to map state between joint and Cartesian space, it shares the properties of
+both `JointState` and `CartesianState`; the accessor methods are listed below:
+
+- `get_name()` returns the "robot name", i.e. the name of the corresponding `JointState`
+- `get_frame()` returns the end-effector frame name, i.e. the name of the corresponding `CartesianState`
+- `get_joint_names()` returns the vector of joint names of the corresponding `JointState`
+- `get_reference_frame()` returns the reference frame of the corresponding `CartesianState`
+
+Each property is initialized on construction and has a corresponding setter for post-construction modifications.
+<!-- TODO(#109): set_frame is not currently implemented -->
```

Comment:
Good find! Do you want to include this in 7.0?

Reply:
To me it was non-blocking but I guess it's a small enough change that we can include it